### ISSUE TITLE
TypeScript parser results in extra indentation if a line starting with a semicolon follows a comment

### ISSUE THREAD

**Issue by clarkdave** [prettier/prettier#5837]
**Prettier 1.16.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzArlMMCW0AEAzjAIYBOMAFAJT7D74A6UDkUx+AJhAOb4C8+KHADu+ACK9KwZgwZQSAWzhJ8AcgCCAGxxg4agDSz8AX2rHjAektcIRGOlSpjAbgDaaspkPqARuQBrNQBdADpUCDIAURIwAAtKAA8BAD46YwZuHjdEsLASLS1KLPMWU1KTEAMQCAAHXGhCZFByMggRAAVyBCaUEgA3CBxOKpBfMliAuBgAZVrYnCgeZBgvOGq4mEUtAHU4nHhCeb0ZnoOcfoOAT2RwQibqxcI4Cg6JnkUSZFQC5+qAK0IiQAQhMwFNZko4AAZRZwb6-dYgQGJGaLHhaOAARXQEHgCK0fxA8zIzzItxgV1qcEIYDIOHqo1q9NgO2GMDiyAAHAAGarMiDPHYTWq3Zk0l79eHVMhwACO6BwsreJA+XyQP0JSOeihwKzW1UI6MxOLx8I1iOqpF8bM4HOQACYrRMcDolgBhCCKT63KDQaUgdDPAAqJF8vU1zxMJiAA)
```sh
# Options (if any):
--parser=typescript
```

**Input:**
```jsx
function start() {  
  const dog = new Dog({
    name: 'Alice',
  })
  
  // do stuff
  ;['run', 'bark'].forEach(x => {
    dog[x].call(dog)
  })
}
```

**Output:**

Note how everything from `name: "Alice"` to the comment is indented an extra 2 spaces

```jsx
function start() {
  const dog = new Dog({
      name: "Alice"
    });

    // do stuff
  ["run", "bark"].forEach(x => {
    dog[x].call(dog);
  });
}

```

**Expected behavior:**

In all other parsers (babel, flow), these lines do not have any extra indentation:

```jsx
function start() {
  const dog = new Dog({
    name: "Alice"
  });

  // do stuff
  ["run", "bark"].forEach(x => {
    dog[x].call(dog);
  });
}
```

**Comment by lydell** [comment]
Formatting once more, the extra indent disappears:

**Prettier 1.16.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzArlMMCW0AEAzjAIYBOMAFAJT7D74A6UDkUx+AJhAOb4C8+KHADu+ACK9KwZgwZQSAWzhJ8AcgCCAGxxg4agDSz8AX2rHjAektcIRGOlSpjAbgDaaspkPqARuQBrNQBdADpUCDIAURIwAAtKAA8BAD46YwZuHjdEsLASLS1KLPMWU1KTEAMQCAAHXGhCZFByMggRAAVyBCaUEgA3CBxOKpBfMliAuBgAZVrYnCgeZBgvOGq4mEUtAHU4nHhCeb0ZnoOcfoOAT2RwQibqxcI4Cg6JnkUSZFQC5+qAK0IiQAQhMwFNZko4AAZRZwb6-dYgQGJGaLHhaOAARXQEHgCK0fxA8zIzzItxgV1qcEIYDIOHqo1q9NgO2GMDiyAAHAAGarMiDPHYTWq3Zk0l79eHVMhwACO6BwsreJA+XyQP0JSOeihwKzW1UI6MxOLx8I1iOqpF8bM4HOQACYrRMcDolgBhCCKT63KDQaUgdDPAAqJF8vU1zxMJiAA)
```sh
--parser typescript
```

**Input:**
```tsx
function start() {  
  const dog = new Dog({
    name: 'Alice',
  })
  
  // do stuff
  ;['run', 'bark'].forEach(x => {
    dog[x].call(dog)
  })
}
```

**Output:**
```tsx
function start() {
  const dog = new Dog({
      name: "Alice"
    });

    // do stuff
  ["run", "bark"].forEach(x => {
    dog[x].call(dog);
  });
}

```

**Second Output:**
```tsx
function start() {
  const dog = new Dog({
    name: "Alice"
  });

  // do stuff
  ["run", "bark"].forEach(x => {
    dog[x].call(dog);
  });
}

```

**Comment by thorn0** [comment]
Fixed by #7094
